v2/dtos/requests/event_test.go

```diff
@@ -21,20 +22,29 @@ var testAddEvent = AddEventRequest{
 	BaseRequest: common.BaseRequest{
 		RequestID: ExampleUUID,
 	},
-	DeviceName: TestDeviceName,
-	Origin:     TestOriginTime,
-	Readings:   nil,
+	Event: dtos.Event{
+		DeviceName: TestDeviceName,
+		Origin:     TestOriginTime,
+		Readings: []dtos.BaseReading{{
+			DeviceName: TestDeviceName,
+			Name:       TestDeviceResourceName,
+			SimpleReading: dtos.SimpleReading{
+				Value:     TestReadingValue,
+				ValueType: TestReadingValueType,
+			},
+		}},
+	},
 }
 
 func TestAddEventRequest_Validate(t *testing.T) {
 	valid := testAddEvent
 	noReID := testAddEvent
 	noReID.RequestID = ""
 	noDeviceName := testAddEvent
-	noDeviceName.DeviceName = ""
+	noDeviceName.Event.DeviceName = ""
 	noDeviceOrigin := testAddEvent
-	noDeviceOrigin.DeviceName = ""
-	noDeviceOrigin.Origin = 0
+	noDeviceOrigin.Event.DeviceName = ""
+	noDeviceOrigin.Event.Origin = 0
```

Comment:
Add tests cases for missing Origin, missing Readings and invalid Reading (missing DeviceName, missing Name & missing  Origin)

Reply:
Added as suggested.